geoh5py/shared/entity.py

```diff
@@ -95,6 +95,8 @@ def __init__(  # pylint: disable=too-many-arguments
         self.on_file = on_file
         self.name = name or self._default_name or type(self).__name__
         self._entity_type = self.validate_entity_type(entity_type)
+        self._parent = None
```

Comment:
it's an attribute for all entities. 
Are we sure it won't break anything in the h5writers?

Reply:
It was already set by the `parent.setter` below. Because I had to move some of the functionality of the setter as a method, then mypy no longer detects that self._parent is an attribute of the class. That's all.